docs: userdata

Therdel opened the pull request:
- address issue #1234 by updating configuration file docs
    - The other undocumented sections [behave.formatters and behave.runners](https://github.com/behave/behave/blob/v1.3.0/behave/configuration.py#L576-L577) remain
- give `Userdata` its own documentation page

jenisys commented:
Thanks, a clear improvement.

BUT: Are you aware that `behave.rst` is generated by [update_behave_rst.py](https://github.com/behave/behave/blob/main/docs/update_behave_rst.py).
There your change will be overwritten/removed on the next run of this update-script.

Therdel commented:
Curious. I did `invoke docs` as written in [contributing.rst](https://github.com/behave/behave/blob/v1.3.0/docs/contributing.rst) in my PR, and my changes persisted 🤔
...
Ah. [docs.py calls sphinx-build](./tasks/docs.py)](https://github.com/behave/behave/blob/v1.3.0/tasks/docs.py#L68) directly, without calling the update script like the [Makefile for Sphinx does](https://github.com/behave/behave/blob/v1.3.0/docs/Makefile#L6).
Is this duplicated docs build code?

I merged some duplicated definitions of excluded config file options and special config sections and moved the Additional configurations section to `behave.rst-template`

jenisys commented:
Two points:
* I have doubts that your changes in `behave.configuration` related to `SPECIAL_CONFIG_SECTIONS` will work for INI files (but I have not checked it yet)
* English: Use uppercase Letters for substantives in section titles

Therdel commented:
putting the PR to Draft.

- fixed the title.
- Also, I think the Userdata doc isn't visible enough in `Noteworthy in Version 1.2.5`, so I refactored it as a main documentation page. Also, I rephrased it and adapted the structure.

jenisys commented:
The failures seems to be caused by the changed if-condition in `Configuration` class constructor: 

```python
# -- APPROX: Line 830ff
if key.startswith("_") or key in CONFIGFILE_EXCLUDED_OPTIONS
# -- MUST BE: Corresponding to logic-expression before
if key.startswith("_") and key not in CONFIGFILE_EXCLUDED_OPTIONS:
```

jenisys commented:
@Therdel 
Thanks for the effort.
